### PR TITLE
feat: Integration with fast_float

### PR DESCRIPTION
**Pull request by swaingotnochill**
See #1069 

This is a draft pull request and the work is in progress.

The goal of this is to get timely feedback and reviews from the reviewers.

TODO:
- [x] Create the wrapper and archive it.
- [] Replace it with strtod.
- [] Benchmark it.
- [] Run it on different platforms.

cc: @madolson @PingXie

### CODE REVIEW

**Comment by madolson** [comment]
@parthpatel FYI.

If possible we would rather not take the C++ dependency, although that might be doable. Do you know how much effort it would be to port the code into C. The templating code seems very generic, but I don't think we need all the flexibility it is providing.

**Comment by swaingotnochill** [comment]
@madolson do you suggest to port the cpp code to c rather than use fast_float?

File: deps/fast_float/fast_float.h [hunk 1]
```diff
@@ -0,0 +1,3869 @@
+// fast_float by Daniel Lemire
```
Comment (parthpatel) [inline deps/fast_float/fast_float.h:1]: if you embed this file, and in future there are bugfixes to this library, how do we track it? is it possible to use submodules for this https://git-scm.com/book/en/v2/Git-Tools-Submodules ?

File: deps/fast_float/fast_float_strtod.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+extern "C"
+{
+    double fast_float_strtod(const char *nptr, char **endptr)
+    {
+        double result;
+        auto answer = fast_float::from_chars(nptr, nptr + strlen(nptr), result);
+
+        if (answer.ec == std::errc())
```
Comment (parthpatel) [inline deps/fast_float/fast_float_strtod.cpp:33]: what is this check doing? shouldn't you just check for null?

File: deps/fast_float/fast_float_strtod.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            if (endptr)
```
Comment (parthpatel) [inline deps/fast_float/fast_float_strtod.cpp:35]: you can move this outside of if block, it is common.

File: deps/fast_float/fast_float_strtod.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ *
+ * @note This function uses the fast_float library (https://github.com/fastfloat/fast_float)
+ *       for the actual conversion, which can be significantly faster than standard library functions.
+ *
+ * @see https://github.com/fastfloat/fast_float for more information on the underlying library.
+ */
+
+extern "C"
+{
+    double fast_float_strtod(const char *nptr, char **endptr)
```
Comment (parthpatel) [inline deps/fast_float/fast_float_strtod.cpp:28]: @madolson do we always use double type in Redis? Would we never want to use float?

**Comment by parthpatel** [comment]
> @parthpatel FYI.
> 
> If possible we would rather not take the C++ dependency, although that might be doable. Do you know how much effort it would be to port the code into C. The templating code seems very generic, but I don't think we need all the flexibility it is providing.

Even if we remove the templating for character type, we still have "class" and "std::" library references. With little more effort, we should be able to remove it. But we won't be able to merge fixes or changes from upstream. In my opinion, we should start with fast_float library as a sub-module (instead of copying the code) and if we run into C++ related issues, then we can still internally change the implementation.

File: deps/fast_float/fast_float.h (hunk 1 above)
Comment (swaingotnochill) [reply deps/fast_float/fast_float.h:1]: The proper scenario is to update the header file with the bug fix version. The diff will be large, but we will be kind of updating the third party dep. However, if we plan to make bug fix, it's not a good idea. Best would be to create a fork and make sure its parallel to the upstream, and we can make custom fixes as well. But it then makes additional overhead.

File: deps/fast_float/fast_float_strtod.cpp (hunk 2 above)
Comment (swaingotnochill) [reply deps/fast_float/fast_float_strtod.cpp:33]: fast_float is written in a way where it returns error and we have to handle them explicitly. That's present in their docs.

**Comment by parthpatel** [comment]
@swaingotnochill I couldn't push changes to your PR, so I created what I was proposing as a draft PR here https://github.com/valkey-io/valkey/pull/1260

**Comment by zuiderkwast** [comment]
Replaced by #1260

**Pull request closed without merging**